colorizer_data/converter.py

```diff
@@ -0,0 +1,389 @@
+import json
+import logging
+import multiprocessing
+import os
+import pathlib
+import time
+from typing import Dict, List, Optional, Sequence, TypedDict, Union
+
+from aicsimageio import AICSImage
+from dataclasses import dataclass
+from pandas import DataFrame
+from pandas.core.groupby.generic import DataFrameGroupBy
+import pandas as pd
+import numpy as np
+
+
+from colorizer_data.types import BackdropMetadata, ColorizerMetadata, FeatureInfo
+from colorizer_data.utils import (
+    INITIAL_INDEX_COLUMN,
+    generate_frame_paths,
+    get_total_objects,
+    remap_segmented_image,
+    sanitize_key_name,
+    sanitize_path_by_platform,
+    scale_image,
+    update_bounding_box_data,
+)
+from colorizer_data.writer import ColorizerDatasetWriter
+
+DEFAULT_OBJECT_ID_COLUMN = "ID"
+DEFAULT_TIMES_COLUMN = "Frame"
+DEFAULT_TRACK_COLUMN = "Track"
+DEFAULT_IMAGE_COLUMN = "File Path"
+DEFAULT_CENTROID_X_COLUMN = "Centroid X"
+DEFAULT_CENTROID_Y_COLUMN = "Centroid Y"
+DEFAULT_OUTLIER_COLUMN = "Outlier"
+
+
+@dataclass
+class ConverterConfig(TypedDict):
+    object_id_column: str
+    times_column: str
+    track_column: str
+    image_column: str
+    centroid_x_column: str
+    centroid_y_column: str
+    outlier_column: str
+    backdrop_columns: Optional[List[str]] = None
+    backdrop_info: Optional[Dict[str, BackdropMetadata]] = None
+    feature_column_names: Optional[List[str]] = None
+    feature_info: Optional[Dict[str, FeatureInfo]] = None
+    use_json: bool = False
+
+
+def _get_image_from_row(row: pd.DataFrame, config: ConverterConfig) -> AICSImage:
+    zstackpath = row[config["image_column"]]
+    zstackpath = sanitize_path_by_platform(zstackpath)
+    return AICSImage(zstackpath)
+
+
+def _make_frame(
```

Comment:
Nearly identical to https://github.com/allen-cell-animated/colorizer-data/blob/main/colorizer_data/bin/example_scripts/convert_nucmorph_data.py#L124-L160